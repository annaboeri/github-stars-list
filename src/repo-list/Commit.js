import React from 'react';

const Commit = props => {
  const { commit } = props.commit;
  const commitDate = new Date(commit.author.date).toLocaleString();

  return (
    <div className="commit">
      <a href={props.commit.html_url}><p><i class="fas fa-history"></i> {commit.message}</p></a>
      <p>{commitDate} - {commit.author.name}</p>
    </div>
  );
}

export default Commit;
