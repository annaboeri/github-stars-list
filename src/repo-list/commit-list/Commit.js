import React from 'react';

const Commit = props => {
  const { commit } = props.commit;
  const commitDate = new Date(commit.author.date).toLocaleTimeString();

  return (
    <li className="commit">
      <a href={props.commit.html_url}>
        <p className="ellipsis commit-message">
          <span>{commit.message}</span>
        </p>
      </a>
      <p>
        <i className="far fa-clock"></i>{commitDate}
        <i className="fas fa-user-circle"></i>{commit.author.name}
      </p>
    </li>
  );
}

export default Commit;
