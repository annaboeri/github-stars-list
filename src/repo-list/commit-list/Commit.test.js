import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Commit from './Commit';

describe('Commit Component', () => {
  it('should render correctly', (done) => {
    const props = {
      commit: {
        html_url: '/',
        commit: {
          message: 'my commit message',
          author: {
            name: 'my name' ,
            date: "2019-12-07T14:47:48Z",
          }
        }
      }
    }
    const component = shallow(<Commit {...props} />);
    const tree = toJson(component);

    expect(tree).toMatchSnapshot();
    expect(component.find('.commit').exists()).toBe(true);
    done();
  });
});
