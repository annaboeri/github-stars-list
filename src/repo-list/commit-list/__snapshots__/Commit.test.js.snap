// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commit Component should render correctly 1`] = `
<li
  className="commit"
>
  <a
    href="/"
  >
    <p
      className="ellipsis commit-message"
    >
      <span>
        my commit message
      </span>
    </p>
  </a>
  <p>
    <i
      className="far fa-clock"
    />
    6:47:48 AM
    <i
      className="fas fa-user-circle"
    />
    my name
  </p>
</li>
`;
